pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -1831,7 +1831,49 @@ public class Foo {
         </example>
     </rule>
 
-    <rule name="UseTryWithResources"
+  <rule name="UseStandardCharsets"
+        language="java"
+        since="6.34"
+        minimumLanguageVersion="1.7"
+        message="Please use StandardCharsets constants"
+        class="net.sourceforge.pmd.lang.rule.XPathRule"
+        externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_bestpractices.html#usestandardcharsets">
+    <description>
+      StandardCharsets has constants for various common character
+      sets, they should be used instead of calling &quot;Charset.forName(name)&quot;
+    </description>
+    <priority>3</priority>
+    <properties>
+      <property name="version" value="2.0"/>
+      <property name="xpath">
+        <value>
+          <![CDATA[
+//PrimaryExpression[PrimaryPrefix/Name/@Image = 'Charset.forName']/PrimarySuffix/Arguments/ArgumentList/Expression/PrimaryExpression/PrimaryPrefix/Literal[@Image = '"US-ASCII"' or @Image = '"ISO-8859-1"'  or @Image = '"UTF-8"'  or @Image = '"UTF-16BE"'  or @Image = '"UTF-16LE"'  or @Image = '"UTF-16"']
```

Comment:
```suggestion
//PrimaryExpression[PrimaryPrefix/Name/@Image = 'Charset.forName']/PrimarySuffix/Arguments/ArgumentList/Expression/PrimaryExpression/PrimaryPrefix/Literal[@Image = ('"US-ASCII"', '"ISO-8859-1"', '"UTF-8"', '"UTF-16BE"',  '"UTF-16LE"', '"UTF-16"')]
```

In XPath 2, `=` tries to make at least a matching pair between an element of the sequence on the left and another element of the sequence on the right. So this can be simplified.